packages/neos-ts-interfaces/src/index.ts

```diff
@@ -0,0 +1,141 @@
+export type NodeContextPath = string;
+export type NodeTypeName = string;
+
+// TODO: for some reason (probably due to immer) I can not use ReadonlyArray here
+export interface NodeChildren extends Array<{
+    contextPath: NodeContextPath;
+    nodeType: NodeTypeName;
+}> {}
+
+export interface NodePolicy extends Readonly<{
+    disallowedNodeTypes: NodeTypeName[];
```

Comment:
Is `ReadonlyArray` not usable here because of `immer`, too?

Reply:
Will check.

Reply:
Yes, it gets me `Type 'NodeMap' is not assignable to type '{ [x: string]: any; }'.
  Index signatures are incompatible.
    Type 'Node' is not assignable to type '{ contextPath: string; name: string; identifier: string; nodeType: string; label: string; isAutoCreated: boolean; depth: number; children: any[]; matchesCurrentDimensions: boolean; properties: any; isFullyLoaded: boolean; uri: string; policy?: any | undefined; }'.
      Types of property 'policy' are incompatible.
        Type 'NodePolicy | undefined' is not assignable to type '{ disallowedNodeTypes: string[]; canRemove: boolean; canEdit: boolean; disallowedProperties: string[]; } | undefined'.
          Type 'NodePolicy' is not assignable to type '{ disallowedNodeTypes: string[]; canRemove: boolean; canEdit: boolean; disallowedProperties: string[]; }'.
            Types of property 'disallowedNodeTypes' are incompatible.
              Type 'ReadonlyArray<string>' is not assignable to type 'string[]'.
                Property 'pop' is missing in type 'ReadonlyArray<string>'.`